tests/scripts/all.sh

```diff
@@ -3363,7 +3363,7 @@ component_build_tfm() {
 
     msg "build: TF-M config, gcc native build"
     make clean
-    make lib CC="gcc" CFLAGS="-Os -std=c99 -Werror -Wall -Wextra -Wwrite-strings -Wpointer-arith -Wimplicit-fallthrough -Wshadow -Wvla -Wformat=2 -Wno-format-nonliteral -Wshadow -Wformat-signedness -Wformat-overflow=2 -Wformat-truncation -Wlogical-op"
```

Comment:
Note that we have GCC 11.2 and Clang 14 on the CI (via the Ubuntu 22.04 image), but you have to arrange for them with a `support_CCC` function that checks the version, because components run on the oldest Linux distribution by default. (Partly because this way we test that we aren't relying on a feature of the newest compilers, and partly due to intertia.)